crates/cli-support/src/js/mod.rs

```diff
@@ -1364,23 +1397,15 @@ impl<'a> Context<'a> {
         let use_encode_into = format!(
             "
                 {}
-                let size = arg.length;
-                let ptr = wasm.__wbindgen_malloc(size);
-                let writeOffset = 0;
-                while (true) {{
-                    const view = getUint8Memory().subarray(ptr + writeOffset, ptr + size);
-                    const {{ read, written }} = cachedTextEncoder.encodeInto(arg, view);
-                    writeOffset += written;
-                    if (read === arg.length) {{
-                        break;
-                    }}
-                    arg = arg.substring(read);
+                if (offset !== arg.length) {{
                     ptr = wasm.__wbindgen_realloc(ptr, size, size += arg.length * 3);
+                    const view = getUint8Memory().subarray(ptr + offset, ptr + size);
+                    offset += cachedTextEncoder.encodeInto(arg, view).written;
```

Comment:
Could a debug assert of some form be included after this to ensure that it wrote everything?

Reply:
We can't *exactly* use `debug_assert` in the JS code, but I guess I could add something if `debug_assertions` is on...

Reply:
There's a `--debug` flag to `wasm-bindgen` itself which can control whether the assertion is emitted or not

Reply:
Example: https://github.com/rustwasm/wasm-bindgen/blob/470eea9/crates/cli-support/src/js/js2rust.rs#L107-L113

Reply:
Ah great.

Reply:
Hmm, I'm not sure what to check here. I mean, I'm passing all that's left of `arg` to `encodeInto`, so it has to be written fully - what condition would I check to prove it?